File: healthchecks/payment-service/src/main/java/com/varunu28/paymentservice/dto/CreatePaymentDto.java
package com.varunu28.paymentservice.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import java.util.UUID;

public record CreatePaymentDto(
    @JsonProperty("idempotency_key") @NotBlank String idempotencyKey,
    @JsonProperty("amount") @Positive double amount,
    @JsonProperty("customer_id") @NotNull UUID customerId) {}

File: saga/payment-service/src/main/kotlin/com/saga/paymentservice/CreatePaymentDto.kt
package com.saga.paymentservice

import com.fasterxml.jackson.annotation.JsonProperty

data class CreatePaymentDto(
    @JsonProperty("order_id") val orderId: String,
    val amount: Double,
    @JsonProperty("customer_name") val customerName: String,
    @JsonProperty("idempotency_key") val idempotencyKey: String,
)
